src/Pester.ps1

```diff
@@ -1126,11 +1126,18 @@ function Invoke-Pester {
 
             # exit with exit code if we fail and even if we succeed, othwerise we could inherit
             # exit code of some other app end exit with it's exit code instead with ours
-            if ($PesterPreference.Run.Exit.Value) {
-                exit ($run.FailedCount + $run.FailedBlocksCount + $run.FailedContainersCount)
+            $failedCount = $run.FailedCount + $run.FailedBlocksCount + $run.FailedContainersCount
+            if ($PesterPreference.Run.Exit.Value -and 0 -ne $failedCount) {
+                # fail with the number of failed tests when there are any
+                # and the exit preference is set. This will fail the run in CI
+                # when any tests failed.
+                exit $failedCount
             }
             else {
-                [System.Environment]::ExitCode = $run.FailedCount + $run.FailedBlocksCount + $run.FailedContainersCount
+                # just set exit code but don't fail when the option is not set
+                # or when there are no failed tests, to ensure that we can run
+                # multiple successful runs of Invoke-Pester in a row.
+                [System.Environment]::ExitCode = $failedCount
```

Comment:
This breaks the original issue. I'm not familiar with `[System.Environment]::ExitCode`, but it doesn't affect `$LASTEXITCODE` which we want to reset. AFAIK we need to either run an application that returns 0 or use `$global:LASTEXITCODE = 0` to reset it in the else-block (or during a teardown-step at the end of each container?).

Demo of original issue (#1616) :
```
$c = [PesterConfiguration]::Default
$c.Run.Scriptblock = {
    Describe "LastExitCode Demos" {
        It 'Works since next legacy app reset the lastexitCode' {
            { cmd /c exit /b 1 } | Should -Not -Throw
            $LASTEXITCODE | Should -Be 1
        }
    }
}
#$c.Run.Exit = $true

Write-Host "Running Tests"
Invoke-Pester -Configuration $c

Write-Host "LastExitCode: $($LASTEXITCODE)"
Write-Host "EnvironmentExitCode: $([Environment]::ExitCode)"
```

Reply:
implemented